Dockerfile

```diff
@@ -70,142 +78,121 @@ RUN sudo apt install -y --no-install-recommends \
+# Install ros-melodic-desktop-full
+RUN sudo apt-get update && sudo apt-get install ros-melodic-desktop-full -y --no-install-recommends
+# Additional Dependencies
+RUN sudo apt-get install ros-melodic-octomap ros-melodic-octomap-mapping ros-melodic-octomap-msgs ros-melodic-octomap-ros ros-melodic-octomap-rviz-plugins ros-melodic-octomap-server ros-melodic-dynamic-edt-3d -y --no-install-recommends
+RUN sudo apt-get install ros-melodic-catkin ros-melodic-teleop-twist-keyboard python-pip python-setuptools python-wstool python-catkin-tools -y --no-install-recommends
+RUN sudo apt-get install ros-melodic-cmake-modules 
+
+
+# ==============================================================
+# TartanAir Codes (https://github.com/Amigoshan/tartanair.git)
+# ==============================================================
+# Clone Repo
+ARG GITHUB_USER=<update>
+ARG GITHUB_PWD=<update>
+ARG WORKSPACE=tartan_ws
 WORKDIR /$FOLDER_NAME/$WORKSPACE
 RUN sudo chown -R $USERNAME /$FOLDER_NAME/$WORKSPACE
-WORKDIR src
-COPY ./tartanair/ /$FOLDER_NAME/$WORKSPACE/src
-
-# (Git clone doesn't work somehow)
-# RUN git clone https://theairlab:airlab123@github.com/Amigoshan/tartanair.git
-# RUN mv tartainair src
+RUN git clone https://$GITHUB_USER:$GITHUB_PWD@github.com/Amigoshan/tartanair.git
+RUN sudo mv /$FOLDER_NAME/$WORKSPACE/tartanair /$FOLDER_NAME/$WORKSPACE/src
 
+# Build Workspace
+# data_type expo_base frontier_base octomap_mapping ompl planner_base roadmap_generator sample_pipeline
 WORKDIR /$FOLDER_NAME/$WORKSPACE
 RUN echo "source /opt/ros/melodic/setup.bash" >> ~/.bashrc
-RUN source /opt/ros/melodic/setup.bash \
-&& sudo -E catkin build
-# data_type expo_base frontier_base octomap_mapping ompl planner_base roadmap_generator sample_pipeline
-
-
-## Source /$FOLDER_NAME/$WORKSPACE/devel/setup.bash automatically
+RUN source /opt/ros/melodic/setup.bash && sudo -E catkin build
 RUN echo "source /$FOLDER_NAME/$WORKSPACE/devel/setup.bash" >> ~/.bashrc
 
-## Mapping Dependencies
-RUN sudo apt update && sudo apt install python-tk python-numba -y --no-install-recommends
-RUN pip install msgpack-rpc-python pyquaternion scipy
-
-## Create directory to store maps
-WORKDIR /$FOLDER_NAME/maps/OccMap
+# Mapping Dependencies
+RUN sudo apt-get update && sudo apt-get install python-tk python-numba -y --no-install-recommends
+RUN pip install pip wheel msgpack-rpc-python pyquaternion scipy
+# TODO: Add packages for python3
 
+# Copy settings_mapping.json to ~Document/AirSim/settings.json
+WORKDIR /home/$USERNAME/Documents/AirSim
+RUN sudo mv /$FOLDER_NAME/$WORKSPACE/src/sample_pipeline/settings_mapping.json /home/$USERNAME/Documents/AirSim/settings.json
```

Comment:
Thanks! May want to leave this out of the dockerfile though, cause the data collection part requires a sample_pipeline/settings_data.json as AirSim/settings.json instead.

Reply:
Agreed, will remove this.

Reply:
Removed.